src/GoTrueClient.ts

```diff
@@ -1156,81 +1165,113 @@ export default class GoTrueClient {
+      if (sessionError) {
+        return { data: null, error: sessionError }
+      }
 
-    const { data, error } = await _request(
-      this.fetch,
-      'POST',
-      `${this.url}/factors/${params.factorId}/verify`,
-      {
-        body: { code: params.code, challenge_id: params.challengeId },
-        headers: this.headers,
-        jwt: sessionData?.session?.access_token,
+      const { data, error } = await _request(
+        this.fetch,
+        'POST',
+        `${this.url}/factors/${params.factorId}/verify`,
+        {
+          body: { code: params.code, challenge_id: params.challengeId },
+          headers: this.headers,
+          jwt: sessionData?.session?.access_token,
+        }
+      )
+      if (error) {
+        return { data: null, error }
       }
-    )
-    if (error) {
-      return { data: null, error }
-    }
 
-    await this._saveSession({
-      expires_at: Math.round(Date.now() / 1000) + data.expires_in,
-      ...data,
-    })
-    this._notifyAllSubscribers('MFA_CHALLENGE_VERIFIED', data)
+      await this._saveSession({
+        expires_at: Math.round(Date.now() / 1000) + data.expires_in,
+        ...data,
+      })
+      this._notifyAllSubscribers('MFA_CHALLENGE_VERIFIED', data)
 
-    return { data, error }
+      return { data, error }
+    } catch (error) {
+      if (isAuthError(error)) {
+        return { data: null, error }
+      }
+      throw error
+    }
   }
 
   /**
    * Creates a challenge which a user can verify against
-   * @param factorID System assigned identifier for authenticator device as returned by enroll
+   * @param factorId System assigned identifier for authenticator device as returned by enroll
    */
   private async _challenge(params: MFAChallengeParams): Promise<AuthMFAChallengeResponse> {
-    const { data: sessionData, error: sessionError } = await this.getSession()
-    if (sessionError) {
-      return { data: null, error: sessionError }
-    }
+    try {
+      const { data: sessionData, error: sessionError } = await this.getSession()
+      if (sessionError) {
+        return { data: null, error: sessionError }
+      }
 
-    return await _request(this.fetch, 'POST', `${this.url}/factors/${params.factorId}/challenge`, {
-      headers: this.headers,
-      jwt: sessionData?.session?.access_token,
-    })
+      return await _request(
+        this.fetch,
+        'POST',
+        `${this.url}/factors/${params.factorId}/challenge`,
+        {
+          headers: this.headers,
+          jwt: sessionData?.session?.access_token,
+        }
+      )
+    } catch (error) {
+      if (isAuthError(error)) {
+        return { data: null, error }
+      }
+      throw error
+    }
   }
+
+  /**
+   * Creates a challenge and immediately verifies it
+   * @param factorId System assigned identifier for authenticator device as returned by enroll
+   * @param code Code Generated by an authenticator device
+   */
```

Comment:
These comments won't work unfortunately because they are on private methods and are not visible from the outside. Move the comment to the `lib/types.ts` file.

Reply:
these are just for internal use since they are private methods, the comments won't be exported anyway